firebase-segmentation/src/main/java/com/google/firebase/segmentation/remote/SegmentationServiceClient.java

```diff
@@ -14,43 +14,160 @@
 public class SegmentationServiceClient {
 
-  private final OkHttpClient httpClient;
-  private final Executor httpRequestExecutor;
+  private static final String FIREBASE_SEGMENTATION_API_DOMAIN =
+      "firebasesegmentation.googleapis.com";
+  private static final String UPDATE_REQUEST_RESOURCE_NAME_FORMAT =
+      "projects/%s/installations/%s/customSegmentationData";
+  private static final String CLEAR_REQUEST_RESOURCE_NAME_FORMAT =
+      "projects/%s/installations/%s/customSegmentationData:clear";
+  private static final String FIREBASE_SEGMENTATION_API_VERSION = "v1alpha";
+
+  private static final String CONTENT_TYPE_HEADER_KEY = "Content-Type";
+  private static final String JSON_CONTENT_TYPE = "application/json";
+  private static final String CONTENT_ENCODING_HEADER_KEY = "Content-Encoding";
+  private static final String GZIP_CONTENT_ENCODING = "gzip";
 
   public enum Code {
     OK,
 
-    SERVER_INTERNAL_ERROR,
+    HTTP_CLIENT_ERROR,
+
+    CONFLICT,
+
+    NETWORK_ERROR,
 
-    ALREADY_EXISTS,
+    SERVER_ERROR,
 
-    PERMISSION_DENIED
+    UNAUTHORIZED,
   }
 
-  public SegmentationServiceClient() {
-    httpClient = new OkHttpClient();
-    httpRequestExecutor = Executors.newFixedThreadPool(4);
+  @NonNull
+  public Code updateCustomInstallationId(
+      long projectNumber,
+      @NonNull String apiKey,
+      @NonNull String customInstallationId,
+      @NonNull String firebaseInstanceId,
+      @NonNull String firebaseInstanceIdToken) {
+    String resourceName =
+        String.format(UPDATE_REQUEST_RESOURCE_NAME_FORMAT, projectNumber, firebaseInstanceId);
+    try {
+      URL url =
+          new URL(
+              String.format(
+                  "https://%s/%s/%s?key=%s",
+                  FIREBASE_SEGMENTATION_API_DOMAIN,
+                  FIREBASE_SEGMENTATION_API_VERSION,
+                  resourceName,
+                  apiKey));
+
+      HttpsURLConnection httpsURLConnection = (HttpsURLConnection) url.openConnection();
+      httpsURLConnection.setDoOutput(true);
+      httpsURLConnection.setRequestMethod("PATCH");
+      httpsURLConnection.addRequestProperty(
+          "Authorization", "FIREBASE_INSTALLATIONS_AUTH " + firebaseInstanceIdToken);
+      httpsURLConnection.addRequestProperty(CONTENT_TYPE_HEADER_KEY, JSON_CONTENT_TYPE);
+      httpsURLConnection.addRequestProperty(CONTENT_ENCODING_HEADER_KEY, GZIP_CONTENT_ENCODING);
+      GZIPOutputStream gzipOutputStream =
+          new GZIPOutputStream(httpsURLConnection.getOutputStream());
+      try {
+        gzipOutputStream.write(
+            buildUpdateCustomSegmentationDataRequestBody(resourceName, customInstallationId)
+                .toString()
+                .getBytes("UTF-8"));
+      } catch (JSONException e) {
+        throw new IllegalStateException(e);
+      } finally {
+        gzipOutputStream.close();
+      }
+      httpsURLConnection.connect();
```

Comment:
I don't think you need `connect()` as you've already written in to the connection